### ISSUE TITLE
[Question]: LangchainEmbedding with huggingfaceEmbeddings vs native llamaindex huggingfaceembedding

### ISSUE THREAD

**Issue by jungwooooo** [run-llama/llama_index#8143]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

Hi. I'm newb on LLM tasks.
I tried to build local LLM system via llamaindex.
but i got difference result between langchain huggingfaceembedding and native huggingfaceembedding.

Here are sample codes.
LanghchainEmbedding with HuggingFaceEmbedding
```
# LlamaIndex with Langchain HuggingFaceEmbedding
#!pip install sentence-transformers
from langchain.embeddings.huggingface import HuggingFaceEmbeddings
from llama_index import LangchainEmbedding, ServiceContext

embed_model = LangchainEmbedding(
  HuggingFaceEmbeddings(model_name="BAAI/bge-small-en-v1.5")
)
embeddings = embed_model.get_text_embedding("Hello World!")
print(len(embeddings))
print(embeddings[:5])

[-0.0032757034059613943, -0.011690760031342506, 0.04155919700860977, -0.03814806044101715, 0.024183105677366257]
```
Native ver.
```
from llama_index.embeddings import HuggingFaceEmbedding

embed_model = HuggingFaceEmbedding(model_name='BAAI/bge-small-en-v1.5')
embeddings = embed_model.get_text_embedding("Hello World!")
print(len(embeddings))
print(embeddings[:5])

[-0.030880559235811234, -0.1102105900645256, 0.3917849361896515, -0.3596276342868805, 0.22797785699367523]
```

As I know, floating error can occur returning slightly different value. but i found that return of native HugEmbedding is average 10 times of return of LangchainEmbedding(wHug).

**Comment by logan-markewich** [comment]
@jungwooooo I could have sworn I confirmed these were equivalent. Will double double check the issue

**Comment by logan-markewich** [comment]
@jungwooooo I stepped through the source code of each

The exact same token IDs get passed to each model. The output of the model itself is different between both libraries, but the post-processing is the same

I'm not sure if sentence-transformers is doing something different to the models when loading, but I can confirm the inputs are the exact same, and the post-processing is the exact same.

**Comment by logan-markewich** [comment]
@jungwooooo spoke too soon, I see sentence transformers has an extra normalize step

**Comment by jungwooooo** [comment]
@logan-markewich Thank you for your kind reply. You said that sentence-transformer has an extra normalize step, so what is the more precise result? native Hug or langchainEmb(w Hug)?

**Comment by logan-markewich** [comment]
@jungwooooo I think in practice both are fairly similar. Normalization helps avoid outliers though.

I have it fixed locally, so now both give the same results. Just deciding on how best to patch this

**Comment by logan-markewich** [comment]
If you are curious, the code looks like this now (with the added normalization)

```
model_output = self._model(**encoded_input)

        if self.pooling == "cls":
            embeddings = self._cls_pooling(model_output)
        else:
            embeddings = self._mean_pooling(
                model_output, encoded_input["attention_mask"]
            )
        
        if self.normalize:
            import torch
            embeddings = torch.nn.functional.normalize(embeddings, p=2, dim=1).tolist()
        
        return embeddings
```

**Comment by jungwooooo** [comment]
@logan-markewich So, is it right that if i use native HugEmb => add normalization step, elif i use langchainemb(w Hug) => use output?

**Comment by jungwooooo** [comment]
@logan-markewich 
i fix this like below code.
```
from llama_index.embeddings import HuggingFaceEmbedding

embed_model = HuggingFaceEmbedding(model_name='BAAI/bge-small-en-v1.5')
embeddings = embed_model.get_text_embedding("Hello World!")
print(len(embeddings))
print(embeddings[:5])
import torch
embeddings = torch.nn.functional.normalize(torch.tensor(embeddings).reshape(1,-1), p=2, dim=1).tolist()
print(embeddings[0][:5])
```
then it return same output value as langchainEmbedding(w Hug)

**Comment by jungwooooo** [comment]
@logan-markewich As i understood, native HuggingFaceEmbedding will need to be updated this normalize issue.

**Comment by logan-markewich** [comment]
@jungwooooo in some initial testing, they still return the same nodes

**Comment by jungwooooo** [comment]
@logan-markewich Thank you for your kind reply. I hope llama_index be the best LLM application package.
And also i will raise another issues for your development!
